lib/src/Base/Algo/openturns/ApproximationAlgorithmImplementation.hxx

```diff
@@ -135,10 +138,10 @@ protected:
 
 private:
   /** Residual */
-  Scalar residual_ = 0.0;
+  Scalar residual_ = -1.0;
```

Comment:
From @regislebrun :

Use -1.0 as the default value as it makes it clear that these quantities haven't been computed yet.
BTW it should be initialized at the hxx level.

Reply:
Fixed.